sklearn/utils/multiclass.py

```diff
@@ -311,7 +246,8 @@ def type_of_target(y):
     if y.dtype.kind == 'f' and np.any(y != y.astype(int)):
         return 'continuous' + suffix
     if len(np.unique(y)) <= 2:
-        assert not suffix, "2d binary array-like should be multilabel"
+        if suffix == "-multioutput":
+            return 'unknown'  # [[1, 2, 3]]
```

Comment:
Does this example pass the `np.unique(y) <= 2` condition? And perhaps from now (or soon) we *do* want this to register as multioutput.